### PR TITLE
03 java service skeleton

### PR DESCRIPTION
**Pull request by Oksana-Shevchenko**

### CODE REVIEW

File: 03-JavaServiceSkeleton/MessageConsumerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                }catch (RemoteException | NullPointerException e) {
+                    Log.e(TAG, "Receive is failed");
+                }
+            }
+        });
+
+        Intent intent = new Intent();
+        intent.setClassName("com.example.shevchenko.message.service", "com.example.shevchenko.message.service.MessageService");
+        try {
+            bindService(intent, serviceConnection, BIND_AUTO_CREATE);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/MessageConsumerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java:57]: Where is unbindService() ?

File: 03-JavaServiceSkeleton/MessageProducerApplication/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 28
+    buildToolsVersion "29.0.3"
+
+    defaultConfig {
+        applicationId "com.example.shevchenko.message"
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/MessageProducerApplication/app/build.gradle]: Did you manage to install 2 different apk with the same applicationId? (Producer, Consumer)

File: 03-JavaServiceSkeleton/MessageProducerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                }catch (RemoteException | NullPointerException e) {
+                    Log.e(TAG, "Send is failed");
+                }
+            }
+        });
+
+        Intent intent = new Intent();
+        intent.setClassName("com.example.shevchenko.message.service", "com.example.shevchenko.message.service.MessageService");
+        try {
+            bindService(intent, serviceConnection, BIND_AUTO_CREATE);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/MessageProducerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java:57]: Where is unbindService() ?

File: 03-JavaServiceSkeleton/MessageService/app/src/main/java/com/example/shevchenko/message/service/MessageService.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    };
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return messageService;
+    }
+
+    @Override
+    public boolean onUnbind(Intent intent) {
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/MessageService/app/src/main/java/com/example/shevchenko/message/service/MessageService.java]: No need to overload the default behavior method

**Review by BulldozerBSG — CHANGES_REQUESTED** [review]

File: 03-JavaServiceSkeleton/MessageConsumerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java (hunk 1 above)
Comment (Oksana-Shevchenko) [reply 03-JavaServiceSkeleton/MessageConsumerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java:57]: Fixed in the commit 6a82d4311abfa610aa43a3dadde40fd9a47a052f

File: 03-JavaServiceSkeleton/MessageProducerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java (hunk 3 above)
Comment (Oksana-Shevchenko) [reply 03-JavaServiceSkeleton/MessageProducerApplication/app/src/main/java/com/example/shevchenko/message/MainActivity.java:57]: Fixed in the commit 6a82d4311abfa610aa43a3dadde40fd9a47a052f

File: 03-JavaServiceSkeleton/MessageService/app/src/main/java/com/example/shevchenko/message/service/MessageService.java (hunk 4 above)
Comment (Oksana-Shevchenko) [reply 03-JavaServiceSkeleton/MessageService/app/src/main/java/com/example/shevchenko/message/service/MessageService.java]: Fixed in the commit 5f20cea483dbfaa98ab31de2ffd831bf3de60bd6

File: 03-JavaServiceSkeleton/MessageProducerApplication/app/build.gradle (hunk 2 above)
Comment (Oksana-Shevchenko) [reply 03-JavaServiceSkeleton/MessageProducerApplication/app/build.gradle]: Thank you! I didn't notice it and this was the reason why I could run only Consumer or Producer app. I have removed the static keyword for the message storage variable in the service as it doesn't need anymore. I can save the state in the String field now.

**Review by BulldozerBSG — APPROVED** [review]
Reviewed. No comments

More details about applicationId and package name:
https://developer.android.com/studio/build/application-id

**Pull request merged**